### PR TITLE
[#62] Wallet details scaffold

### PR DESCRIPTION
**Pull request by ccjernigan**
Implements a details screen showing various wallet addresses

Some known issues requiring followup in future pull requests:
 - #159 Colors for light theme have not been defined yet
 - #160 Color blocks have 2x borders along the top/bottom of the middle item
 - #161 Some address types display placeholder values

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
- [x] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Manual tests: Did you update the [manual tests](../blob/main/docs/testing/manual_testing) as appropriate? _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed._
- [x] Documentation: Did you update documentation as appropriate? (e.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes?
- [x] Screenshots: Did you provide before and after UI screenshots in the description of this pull request?  _This is only applicable for changes that modify the UI._
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place._
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

light
dark

### CODE REVIEW

File: sdk-ext-lib/src/main/java/cash/z/ecc/sdk/model/WalletAddresses.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        suspend fun new(persistableWallet: PersistableWallet): WalletAddresses {
+            // Dispatcher needed because SecureRandom is loaded, which is slow and performs IO
+            // https://github.com/zcash/kotlin-bip39/issues/13
+            val bip39Seed = withContext(Dispatchers.IO) {
+                Mnemonics.MnemonicCode(persistableWallet.seedPhrase.joinToString()).toSeed()
+            }
+
+            // Dispatchers needed until an SDK is published with the implementation of
+            // https://github.com/zcash/zcash-android-wallet-sdk/issues/269
+            val viewingKey = withContext(Dispatchers.IO) {
```
Comment (ccjernigan) [inline sdk-ext-lib/src/main/java/cash/z/ecc/sdk/model/WalletAddresses.kt:29]: @pacu Can you help me confirm that for the addresses supported by the SDK, that I got the right values here?

There are methods for both `deriveUnifiedViewingKeys` and `deriveViewingKeys` for example, and it isn't clear to me which is the right one to call in this situation.

https://github.com/zcash/zcash-android-wallet-sdk/blob/6968797092487b70fe5629a4a0ad11064715d08f/sdk-lib/src/main/java/cash/z/ecc/android/sdk/tool/DerivationTool.kt

Comment (pacu) [reply sdk-ext-lib/src/main/java/cash/z/ecc/sdk/model/WalletAddresses.kt:29]: Well for this situation you would have to access every component individually. 

The concept of UVKs that we have on the SDK now is different as the one that it's going to have later. Then there were no UVKs we actually kind of coined that term ourselves by grouping an IVK and a transparent public key.

File: sdk-ext-lib/src/main/java/cash/z/ecc/sdk/fixture/WalletAddressesFixture.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    const val UNIFIED = "Unified GitHub Issue #161"
+    const val SHIELDED_ORCHARD = "Shielded Orchard GitHub Issue #161"
+    @Suppress("MaxLineLength")
+    const val SHIELDED_SAPLING = "ztestsapling1475xtm56czrzmleqzzlu4cxvjjfsy2p6rv78q07232cpsx5ee52k0mn5jyndq09mampkgvrxnwg"
+    const val TRANSPARENT = "tmXuTnE11JojToagTqxXUn6KvdxDE3iLKbp"
+    const val VIEWING_KEY = "03feaa290589a20f795f302ba03847b0a6c9c2b571d75d80bc4ebb02382d0549da"
+
+    fun new(
+        unified: String = UNIFIED,
+        shieldedOrchard: String = SHIELDED_ORCHARD,
```
Comment (pacu) [inline sdk-ext-lib/src/main/java/cash/z/ecc/sdk/fixture/WalletAddressesFixture.kt:17]: Note: There's no human readable Shielded Orchard Address

**Review by pacu — APPROVED** [review]
Beware of Orchard Address! left comment

File: sdk-ext-lib/src/main/java/cash/z/ecc/sdk/fixture/WalletAddressesFixture.kt (hunk 2 above)
Comment (ccjernigan) [reply sdk-ext-lib/src/main/java/cash/z/ecc/sdk/fixture/WalletAddressesFixture.kt:17]: Will a human readable shielded orchard address be added to the SDK in the future?

Display of a shielded orchard address is in the design, so I'm trying to determine whether we should update the design to remove this or whether this gets resolved in #161.

**Pull request merged**